src/main/java/com/customer/service/CustomerServiceImpl.java

```diff
@@ -9,69 +10,60 @@
 import org.springframework.stereotype.Service;
 
 import com.customer.CustomerResponse.CustomerResponse;
-import com.customer.customerModel.Customer;
-import com.customer.customerRepo.CustomerRepository;
+import com.customer.model.Customer;
+import com.customer.repository.CustomerRepository;
+import com.customer.sorting.SortingComparator;
 
 @Service
 public class CustomerServiceImpl implements CustomerService {
 
+	private CustomerRepository customerRepo;
+
 	@Autowired
-	CustomerRepository customerRepo;
+	CustomerServiceImpl(CustomerRepository customerRepo) {
+		this.customerRepo = customerRepo;
+	}
+
 	@Value("${spring.customer.expDate}")
-	private int expdate;
+	private int expDate;
 
 	@Override
-	public void createCustomer(Customer cust) {
-		cust.setExpiryDate(expdate);
-		customerRepo.save(cust);
+	public Customer createCustomer(Customer cust) {
+		cust.setExpiryDate(expDate);
+		return customerRepo.save(cust);
 	}
 
 	@Override
 	public List<CustomerResponse> getAllCustomers() {
-		List<Customer> cust = customerRepo.findAll();
-		List<CustomerResponse> custResp = cust.stream().map(cus -> {
-			CustomerResponse customerResponse = CustomerResponse.builder().id(cus.getId()).firstName(cus.getAddress())
-					.lastName(cus.getLastName()).address(cus.getAddress()).joiningDate(cus.getJoiningDate())
-					.expiryDate(cus.getExpiryDate()).build();
+		List<Customer> custList = customerRepo.findAll();
+		List<CustomerResponse> custRespList = custList.stream().map(cus -> {
+			CustomerResponse customerResponse = CustomerResponse.builder().id(cus.getId()).firstName(cus.getFirstName())
+					.lastName(cus.getLastName()).joiningDate(cus.getJoiningDate()).expiryDate(cus.getExpiryDate())
+					.address(cus.getAddress())
+					.build();
 			customerResponse.setStatus(cus.getJoiningDate(), cus.getExpiryDate());
 			return customerResponse;
 		}).collect(Collectors.toList());
-		return custResp;
-	}
-
-	@Override
-	public Optional<Customer> findCustomerById(int id) {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
-	@Override
-	public void deleteCustomerById(int id) {
-		// TODO Auto-generated method stub
-
-	}
-
-	@Override
-	public void updateCustomer(Customer cust) {
-		// TODO Auto-generated method stub
-
+		Collections.sort(custRespList, new SortingComparator());
+		return custRespList;
 	}
 
 	@Override
-	public void deleteAllCustomers() {
+	public Object findCustomerById(int id) {
```

Comment:
Use @ControllerAdvise and learn Exceptional handling in spring boot

Reply:
done